osu.Framework.Android/AndroidGameSurface.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using Android.Content;
+using Android.Runtime;
+using Org.Libsdl.App;
+using osu.Framework.Graphics;
+using osu.Framework.Platform;
+using osuTK.Android;
+using osu.Framework.Bindables;
+using Android.Views;
+using AndroidX.Core.View;
+using AndroidX.Window.Layout;
+
+namespace osu.Framework.Android
+{
+    internal class AndroidGameSurface : SDLSurface
+    {
+        private AndroidGameActivity activity { get; } = null!;
+
+        public BindableSafeArea SafeAreaPadding { get; } = new BindableSafeArea();
+
+        public AndroidGameSurface(AndroidGameActivity activity, Context? context)
+            : base(context)
+        {
+            init();
+            this.activity = activity;
+        }
+
+        protected AndroidGameSurface(IntPtr javaReference, JniHandleOwnership transfer)
+            : base(javaReference, transfer)
+        {
+            init();
+        }
+
+        private void init()
+        {
+            if (OperatingSystem.IsAndroidVersionAtLeast(26))
+            {
+                // disable ugly green border when view is focused via hardware keyboard/mouse.
+                DefaultFocusHighlightEnabled = false;
+            }
+        }
+
+        private volatile bool isSurfaceReady;
+
+        public bool IsSurfaceReady => isSurfaceReady;
+
+        public override void HandlePause()
+        {
+            base.HandlePause();
+            isSurfaceReady = false;
+        }
+
+        public override void HandleResume()
+        {
+            base.HandleResume();
+            isSurfaceReady = true;
+        }
+
+        public override WindowInsets? OnApplyWindowInsets(WindowInsets? insets)
+        {
+            updateSafeArea(insets);
+            return base.OnApplyWindowInsets(insets);
+        }
+
+        /// <summary>
+        /// Updates the <see cref="IWindow.SafeAreaPadding"/>, taking into account screen insets that may be obstructing this <see cref="AndroidGameView"/>.
```

Comment:
Should be updated to mention the proper class, and remove `using osuTK.Android;`

```suggestion
        /// Updates the <see cref="IWindow.SafeAreaPadding"/>, taking into account screen insets that may be obstructing this <see cref="AndroidGameSurface"/>.
```